### ISSUE TITLE
hotswap discards non-table values

### ISSUE THREAD

**Issue by notwa** [rxi/lume#18]
https://github.com/rxi/lume/blob/da0d1bb/lume.lua#L694-L695
```lua
    local newmod = require(modname)
    if type(oldmod) == "table" then update(oldmod, newmod) end
```
currently, if the returned value of a module isn't a table, it (newmod) is discarded.

in my case, my module returns a function, and i expect `package.loaded` to contain the new value. i am using `lurker.lua` to trigger hotswapping.
```lua
local some_state = 'blah'
local function exec_stuff(name)
  local f = require('stuff_'..name) -- bug: never returns new function value
  some_state = f(some_state)
end
```

this dirty hack seems to work:
```diff
-    if type(oldmod) == "table" then update(oldmod, newmod) end
+    if type(oldmod) == "table" then update(oldmod, newmod) else oldmod = newmod end
```

**Comment by Alloyed** [comment]
The problem here is that there's no way to modify existing functions: if you've required and stored the same function somewhere else in the past, that version doesn't get updated with your patch, which means hotswap doesn't actually hotswap anything in that case.

A workaround could be to use __call like so:
```lua
return setmetatable({}, {__call = drop_arg(1, my_function)})
```
This is a normal, modifiable table object, so it should work with hotswap, but you can call it as if it were a normal function. That said, if the current behavior is fine with you you can do the same thing manually, without modifying hotswap:
```lua
package.loaded[my_mod] = nil
local f = require(my_mod)
```

**Comment by notwa** [comment]
right, afaik you can only emulate updating existing references either by utilizing a wrapper table or by replacing all of an existing table's values. that's why, in my example code, i'm running `require` each time a function is needed instead of caching it locally. i was expecting this.

what i'm really trying to describe is that i figured hotswap would update non-table *return* values in `package.loaded`. this includes returned numeric and userdata values as well.

nulling out the key in `package.loaded` would work, but that would need to be part of a `lurker.lua` hook to prevent reloading an untouched file, which might be expensive if the value is needed frequently.
